options/rtld/generic/linker.cpp

```diff
@@ -795,6 +816,163 @@ void ObjectRepository::_parseDynamic(SharedObject *object) {
 	}
 }
 
+void ObjectRepository::_parseVerdef(SharedObject *object) {
+	if(!object->versionDefinitionTableOffset) {
+		if(verbose)
+			mlibc::infoLogger()
+				<< "mlibc: Object " << object->name
+				<< " defines no versions" << frg::endlog;
+		return;
+	}
+
+	if(verbose)
+		mlibc::infoLogger()
+			<< "mlibc: Object " << object->name
+			<< " defines " << object->versionDefinitionCount
+			<< " version(s)" << frg::endlog;
+
+	uintptr_t address =
+		object->baseAddress
+		+ object->versionDefinitionTableOffset;
+
+	for(size_t i = 0; i < object->versionDefinitionCount; i++) {
+		elf_verdef def;
+		memcpy(&def, reinterpret_cast<void *>(address), sizeof(elf_verdef));
+
+		// Required by spec.
+		__ensure(def.vd_version == 1);
+		__ensure(def.vd_cnt >= 1);
+		// TODO(qookie): Handle weak versions.
+		__ensure(!(def.vd_flags & ~VER_FLG_BASE));
+
+		// TODO(qookie): Figure out what to do about VERDEF dependency names.
+		// They don't make much sense to me, and it seems to work if we just ignore them.
+		elf_verdaux aux;
+		memcpy(&aux, reinterpret_cast<void *>(address + def.vd_aux), sizeof(elf_verdaux));
+
+		const char *name =
+			static_cast<const char *>(
+				reinterpret_cast<void *>(
```

Comment:
Can't you reinterpret_cast directly to `const char *`?

Reply:
That's true. For some reason I thought you could only `reinterpret_cast` integers into `void *`...

Reply:
I believe that's true of `static_cast`